mmv1/third_party/terraform/services/storage/iam_storage_managed_folder.go

```diff
@@ -0,0 +1,182 @@
+package storage
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"google.golang.org/api/cloudresourcemanager/v1"
+
+	"github.com/hashicorp/terraform-provider-google/google/tpgiamresource"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+var StorageManagedFolderIamSchema = map[string]*schema.Schema{
+	"bucket": {
+		Type:     schema.TypeString,
+		Required: true,
+		ForceNew: true,
+	},
+	"managed_folder": {
+		Type:             schema.TypeString,
+		Required:         true,
+		ForceNew:         true,
+		DiffSuppressFunc: tpgresource.CompareSelfLinkOrResourceName,
```

Comment:
This should also have a validation func that has the same logic as the [name field on the parent resource](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10786-old..auto-pr-10786#diff-fe97e5996e9605a4951a42ccce2c50b9067a734e7e98ccbf98586099062f50a6R61):

```suggestion
		DiffSuppressFunc: tpgresource.CompareSelfLinkOrResourceName,
		ValidateFunc: verify.ValidateRegexp(`/$`),
```

This will require also importing `"github.com/hashicorp/terraform-provider-google-beta/google-beta/verify"`

Reply:
Done